src/ProgressiveCurve.sol

```diff
@@ -82,17 +82,18 @@ contract ProgressiveCurve is BaseCurve {
     /// @dev or to say that another way:
     /// $$\text{shares} = \sqrt{s^2 + \frac{2a}{m}} - s$$
     function previewDeposit(uint256 assets, uint256, /*totalAssets*/ uint256 totalShares)
-        public
```

Comment:
3.20 - Functions in Curve Contracts Could Be external Instead of public

Made all of these `external`